javascript-bank/account.js

```diff
@@ -0,0 +1,39 @@
+/* exported Account */
+function Account(number, holder){
+  this.number = number;
+  this.holder = holder;
+  this.transactions = [];
+  this.balance = 0;
+}
+
+Account.prototype.deposit = function (amt) {
+  if ((amt > 0) && ((amt % 1) === 0)) {
+    let trans = new Transaction('deposit', amt);
+    this.transactions.push(trans);
+    this.balance += amt;
+    this.number += amt
+    return true
+  } else {
+    return false
+  }
+};
+
+Account.prototype.withdraw = function (amt) {
+  if ((amt > 0) && ((amt % 1) === 0)) {
+    let trans = new Transaction('withdrawal', amt);
+    this.transactions.push(trans);
+    this.balance -= amt;
+    this.number -= amt;
+    return true
+  } else {
+    return false
+  }
+};
+
+Account.prototype.getBalance = function () {
+  return this.balance
+};
```

Comment:
The balance should be calculated in the `getBalance` method, not beforehand.

---

![image](https://user-images.githubusercontent.com/16143771/123365410-16565a00-d52b-11eb-952e-6cefa5060cb1.png)